### PR TITLE
Converge branches

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

Converge test suite code for head, 4.1, and 4.0.

## Links

Manager-4.0:  SUSE/spacewalk#13520
Manager-4.1: SUSE/spacewalk#13487

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

### CODE REVIEW

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (srbarrios) [inline testsuite/features/qam/reposync/srv_sync_qam_products.feature:130]: Fine.. just to have a mirror of 4.1 for next 4.2.
But ofc, this will not apply to Uyuni

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:882]: I wonder if we bootstrap the same repo name in Uyuni 🤔

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:1468]: Please don't use 'product' as variable, it is issue to mix with '$product'

Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:1468]: Also, review if for Uyuni, we use a different name like Uyuni-Proxy

File: testsuite/features/step_definitions/navigation_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/navigation_steps.rb:822]: My bad thanks to port it :)

File: testsuite/features/support/env.rb
Comment (srbarrios) [inline testsuite/features/support/env.rb:35]: Thanks!

File: testsuite/features/step_definitions/command_steps.rb
Comment (lkotek) [reply testsuite/features/step_definitions/command_steps.rb:1468]: @srbarrios Very good catch, thanks! I didn't realize that, but yes, you're completely right.

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (Bischoff) [reply testsuite/features/qam/reposync/srv_sync_qam_products.feature:130]: We could also switch to 4.2 here - it's head branch after all. And it will soon come to existence as we are at alpha 1.

Tell me if you prefer that.

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:882]: Don't know.

Apart from that, same as above, we could also start using 42.

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:1468]: I'm not the author of this code.

If you would like such changes, please contact the original authors, and let's do it in separate PRs.

This is only about aligning the code in every branch.

File: testsuite/features/step_definitions/navigation_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/navigation_steps.rb:822]: No worry, you're not the only one to have forgotten ports.

That's the whole purpose of this operation.

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (srbarrios) [reply testsuite/features/qam/reposync/srv_sync_qam_products.feature:130]: We need to have here whatever "Head" version shown in the WebUI, as it shows 4.2, we should have here 4.2.
BUT keep in mind that we will not execute this code either for Uyuni or Head. As the purpose of this code is only for Maintenance Updates.

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:882]: Yes. For head code, this is a typo. Still we need to distinguish products.
Pinging our proxy master, @lkotek

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:1468]: > I'm not the author of this code.
> 
> If you would like such changes, please contact the original authors, and let's do it in separate PRs.
> 
> This is only about aligning the code in every branch.

@Bischoff Do you want to merge something wrong? In Uyuni we need to distinguish with Head. This doesn't apply to spacewalk. So please don't merge it, to fix this wrong code in a future PR.

**Review by srbarrios — CHANGES_REQUESTED** [review]
Before merging this, we need to refactor the ported code, related with proxy, that must have a distinction between uyuni and susemanager products.

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (Bischoff) [reply testsuite/features/qam/reposync/srv_sync_qam_products.feature:130]: I know that. The thing is, this code will eventually become the 4.2 QAM code.

I'll change 4.1 to 4.2 in head.

Comment (Bischoff) [reply testsuite/features/qam/reposync/srv_sync_qam_products.feature:130]: Done.

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:882]: I switched to 4.2.

Thanks for the ping. We can fix the issue here, or in a separate PR, whatever makes more sense for you guys.

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:1468]: We can also fix it as part of this PR, yes, if it is short.

It is just that I would like to focus on synchronizing the branches.

If we find errors at this occasion, fine, but maybe it is cleaner and simpler to do it in separate PRs. That's all.

File: testsuite/features/step_definitions/command_steps.rb
Comment (lkotek) [reply testsuite/features/step_definitions/command_steps.rb:882]: I will take a look. I have a lot of notes about it from few months ago :-)

Comment (lkotek) [reply testsuite/features/step_definitions/command_steps.rb:882]: @srbarrios In case of uyuni we create same bootstrap repo as for Leap 15.2. Proxy specific things needs to be part of activation key, but there is nothing specific to be done at level of bootstrap repo. (Basically uyuni works like SUMA 3.2 and older.)

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:1468]: If we do in a separate PR better, but then let's not port any line at that section.

Comment (lkotek) [reply testsuite/features/step_definitions/command_steps.rb:1468]: @srbarrios I would like to do it in separate PR. TBH I am afraid we can jump into a lot of unexpected consequences as we entering area, where are differences between uyni, head and SUMA itself. I am author of this code, so I volunteer to fix it. But my suggestion is not to do it today on Friday before Christmas. I can create card for it so this won't be forgotten.

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:882]: So for Uyuni we will also use `SUMA-41-PROXY-#{arch}` this on the bootstrap repo command ?

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:1468]: > If we do in a separate PR better, but then let's not port any line at that section.

OK. I will not port anything then. Same for the other one above.

@lkotek i create a card for you in January?

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:1468]: I just realized this code is used in QAM only - no uyuni then, it will become used in 4.2 where we have SUSE Manager only.

But... as we might want to start using QAM test suite at some point for milestones as well, better have so bullet-proof code.

File: testsuite/features/step_definitions/command_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:882]: I will remove the change and let Lukas address this in a separate PR.

**Review by srbarrios — APPROVED** [review]
LGTM. As far we address in a different PR the port related with proxy and the distinction by product.

**Comment by srbarrios** [comment]
Removing approval :P I will take a look when it's not more a WIP hehe

**Comment by Bischoff** [comment]
> Removing approval :P I will take a look when it's not more a WIP hehe

it should not be long now

**Comment by srbarrios** [comment]
Those PRs seems to have extra changes, aside from cherry-picks of missing ports.
Be sure that those changes are also synced ;)

**Comment by Bischoff** [comment]
> Those PRs seems to have extra changes, aside from cherry-picks of missing ports.
> Be sure that those changes are also synced ;)

They are (I port them directly at fixing time, and double-check with a git diff between branches).

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (srbarrios) [inline testsuite/features/qam/reposync/srv_sync_qam_products.feature:124]: 👍

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/qam/reposync/srv_sync_qam_products.feature
Comment (Bischoff) [reply testsuite/features/qam/reposync/srv_sync_qam_products.feature:124]: Yeah, it's a bit of anticipating the future, I know...

**Pull request merged**
